0005-cmx2cdp-protocol/README.md

```diff
@@ -0,0 +1,207 @@
+---
+name: Communication protocol between CMX and CDP services
+shortname: 5/CMX2CDP
+status: raw
+editor: Andrew Pasika <apasika@cybervisiontech.com>
+---
+
+## Introduction
+
+The CMX2CDP protocol is a [Kaa BB: messaging IPC](/0003-messaging-ipc/README.md) protocol extension.
+
+It is intended to solve the problem of communication between CMX and CDP.
+
+## Requirements and constraints
+
+#### Problems and possible solutions
+
+1. _Configuration delivery confirmation._
+
+    Solutions:
+    - CMX should broadcast event that particular endpoint has updated its configuration to specific version.
+
+2. _Configuration push._
+
+    Solutions:
+    - CPD should broadcast event that new configuration is available.
+
+3. _It is possible that there are two CPD instances configured to work with different 
+content types and one originator service instance. When originator service publishes config event 
+message with certain content type it is consumed by both CDPs._
+
+    Solutions:
+    - CDP should filter messages by their content types.
+
+4. If endpoint asks for its configuration but there is no configuration currently available, 
+CDP should somehow inform endpoint about it.
+
+    Solutions:
+    - Use status codes which encapsulate meaningful information: 0 - OK, 1 - FAIL, 2 - NO CONTENT.
+
+#### Key concept
+
+There are two messaging approaches used in CMX-CDP communication:
+* [Targeted messaging](#targeted-messaging)
+* [Broadcast messaging](#broadcast-messaging)
```

Comment:
I don't sure that cmx2cdp protocol won't be extended, so your suggestion can't take place here since, for example, _"Targeted messaging"_ could be used not only for _"Configuration pull"_. Therefore, from my perspective, that is fine now.